pkg/controllers/provisioning/provisioner_test.go

```diff
@@ -1358,4 +1373,363 @@ var _ = Describe("Provisioner/Provisioning", func() {
 			})
 		})
 	})
+
+	Context("Resource Calculations", func() {
+
+		It("should schedule based on the sum of containers and sidecarContainers", func() {
+			pod := test.Pod(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2")},
+				},
+
+				InitContainers: []v1.Container{
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+						},
+					},
+				},
+			})
+			podResources := resources.Ceiling(pod)
+			Expect(podResources.Limits.Cpu().CmpInt64(6)).To(Equal(0))
+			Expect(podResources.Limits.Memory().CmpInt64(3221225472)).To(Equal(0))
+		})
+
+		It("should schedule based on the sum of containers , sidecarContainers and overhead with max of initcontainers", func() {
+			pod := test.Pod(test.PodOptions{
+				Overhead: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("5"),
+					v1.ResourceMemory: resource.MustParse("500Mi"),
+				},
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2")},
+				},
+
+				InitContainers: []v1.Container{
+					{
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+						},
+					},
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("3"), v1.ResourceMemory: resource.MustParse("3Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+						},
+					},
+				},
+			})
+			podResources := resources.Ceiling(pod)
```

Comment:
Same comment here on what this is testing. This is exactly the same test as is in the `nodepool_test.go` file, so there isn't anything different with how this is getting handled. We should test the schedulability of the Provisioner given the available instance types and the pods that contain these native sidecar containers